zebra-state/src/service/check.rs

```diff
@@ -80,23 +80,25 @@ where
         .expect("valid blocks have a coinbase height");
     check::height_one_more_than_parent_height(parent_height, semantically_verified.height)?;
 
+    // skip this check during tests if we don't have enough blocks in the chain
+    // process_queued also checks the chain length, so we can skip this assertion during testing
+    // (tests that want to check this code should use the correct number of blocks)
+    //
+    // TODO: accept a NotReadyToBeCommitted error in those tests instead
+    #[cfg(test)]
     if relevant_chain.len() < POW_ADJUSTMENT_BLOCK_SPAN {
-        // skip this check during tests if we don't have enough blocks in the chain
-        // process_queued also checks the chain length, so we can skip this assertion during testing
-        // (tests that want to check this code should use the correct number of blocks)
-        //
-        // TODO: accept a NotReadyToBeCommitted error in those tests instead
-        #[cfg(test)]
         return Ok(());
+    }
 
-        // In production, blocks without enough context are invalid.
-        //
-        // The BlockVerifierRouter makes sure that the first 1 million blocks (or more) are
-        // checkpoint verified. The state queues and block write task make sure that blocks are
-        // committed in strict height order. But this function is only called on semantically
-        // verified blocks, so there will be at least 1 million blocks in the state when it is
-        // called. So this error should never happen.
-        #[cfg(not(test))]
+    // In production, blocks without enough context are invalid.
+    //
+    // The BlockVerifierRouter makes sure that the first 1 million blocks (or more) are
+    // checkpoint verified. The state queues and block write task make sure that blocks are
+    // committed in strict height order. But this function is only called on semantically
+    // verified blocks, so there will be at least 1 million blocks in the state when it is
+    // called. So this error should never happen on Mainnet or the default Testnet.
+    #[cfg(not(test))]
+    if relevant_chain.is_empty() {
```

Comment:
> The docs for this fn say it panics when "the state contains less than 28 (POW_ADJUSTMENT_BLOCK_SPAN) blocks", but I don't see the panic.

I don't think I've seen it panic either, it may have just returned an error from the `submitblock` RPC method. We may want to update the comment.

Reply:
Ah, I think `POW_ADJUSTMENT_BLOCK_SPAN` is defined as PoWAveragingWindow + PoWMedianBlockSpan here (page 132 of the spec): 
`ActualTimespan(height : N) := MedianTime(height) − MedianTime(height − PoWAveragingWindow)`
`MedianTime(height : N) := median([[ nTime(𝑖) for 𝑖 from max(0, height − PoWMedianBlockSpan) up to height − 1 ]])`

MedianTime has a floor of 0, so, I think we're okay to use this for Mainnet and the default Testnet too.

Reply:
> Yea, let's remove that bit. I guess this PR is a good place to do that.

Ah, it would've panicked at the time that the doc was written because it was using an assertion instead of returning an error.

Reply:
Removed the incorrect docs about the function panicking and added a comment about why it's okay to use a relevant chain that's shorter than the `POW_ADJUSTMENT_BLOCK_SPAN` in https://github.com/ZcashFoundation/zebra/commit/2ad072a19f1713090bbbaccabb5d3a7d381a55f1.